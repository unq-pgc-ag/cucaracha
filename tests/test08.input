fun func(v : Vec) {
  v[1 + 1] := v[2 + 2]
}

fun main() {
  func([1,
        2+2,
        3+3+3,
        4+4+4+4,
        5+5+5+5+5])
}
