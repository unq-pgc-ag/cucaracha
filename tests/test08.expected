(Program
  (Function
    func
    Unit
    (Parameter
      v
      Vec
    )
    (Block
      (StmtVecAssign
        v
        (ExprAdd
          (ExprConstNum
            1
          )
          (ExprConstNum
            1
          )
        )
        (ExprVecDeref
          v
          (ExprAdd
            (ExprConstNum
              2
            )
            (ExprConstNum
              2
            )
          )
        )
      )
    )
  )
  (Function
    main
    Unit
    (Block
      (StmtCall
        func
        (ExprVecMake
          (ExprConstNum
            1
          )
          (ExprAdd
            (ExprConstNum
              2
            )
            (ExprConstNum
              2
            )
          )
          (ExprAdd
            (ExprAdd
              (ExprConstNum
                3
              )
              (ExprConstNum
                3
              )
            )
            (ExprConstNum
              3
            )
          )
          (ExprAdd
            (ExprAdd
              (ExprAdd
                (ExprConstNum
                  4
                )
                (ExprConstNum
                  4
                )
              )
              (ExprConstNum
                4
              )
            )
            (ExprConstNum
              4
            )
          )
          (ExprAdd
            (ExprAdd
              (ExprAdd
                (ExprAdd
                  (ExprConstNum
                    5
                  )
                  (ExprConstNum
                    5
                  )
                )
                (ExprConstNum
                  5
                )
              )
              (ExprConstNum
                5
              )
            )
            (ExprConstNum
              5
            )
          )
        )
      )
    )
  )
)
